### ISSUE TITLE
local storage

### ISSUE THREAD

**Issue by nicolad** [marinanicolai/unsplash-hooks-portofolio#12]

**Comment by nicolad** [comment]
```mermaid
graph LR
A[Browser] -- Stores Data --> B[localStorage]
B -- Key-Value Pairs --> C{Data}
```

Explanation:
- The diagram depicts a relationship between the "Browser" and the `localStorage` object.
- The "Browser" represents the web browser environment where the JavaScript code is executed.
- The `localStorage` object is a built-in feature of web browsers that allows storing data on the client-side.
- The arrow pointing from "Browser" to `localStorage` represents the act of storing data in the `localStorage` object.
- The `localStorage` object stores data in the form of key-value pairs.
- The arrow pointing from `localStorage` to the curly brace `{Data}` signifies that the `localStorage` object can hold various types of data.
- The curly brace `{Data}` represents the stored data, which can be accessed and retrieved using specific keys.

In summary, the diagram illustrates how the `localStorage` object in the browser environment enables the storage of data using key-value pairs. The stored data can later be accessed and retrieved based on the keys provided.

**Comment by nicolad** [comment]
To store and retrieve objects in `localStorage`, you can follow these steps in JavaScript:

Storing an object in `localStorage`:
1. Convert the object to a JSON string using `JSON.stringify()`.
2. Use the `setItem()` method of the `localStorage` object to store the JSON string.
   ```javascript
   const objectToStore = { key: 'value', foo: 'bar' };
   const jsonStr = JSON.stringify(objectToStore);
   localStorage.setItem('myObject', jsonStr);
   ```

Retrieving an object from `localStorage`:
1. Use the `getItem()` method of the `localStorage` object to retrieve the JSON string.
2. If the JSON string exists, parse it back into an object using `JSON.parse()`.
   ```javascript
   const jsonStr = localStorage.getItem('myObject');
   if (jsonStr) {
     const retrievedObject = JSON.parse(jsonStr);
     console.log(retrievedObject);
   }
   ```

Note: When retrieving the object from `localStorage`, make sure to handle cases where the stored value may be null or undefined. Also, be cautious with storing sensitive or large objects in `localStorage` as it has limitations on storage capacity and is accessible to all scripts running in the same origin.
